cvat-sdk/cvat_sdk/core/proxies/tasks.py

```diff
@@ -132,27 +132,25 @@ def upload_data(
             self._client.logger.info("Awaiting for task %s creation...", self.id)
             while True:
                 sleep(status_check_period)
-                (status, response) = self.api.retrieve_status(self.id)
+                requests, response = self._client.api_client.requests_api.list(
+                    action="create", task_id=self.id
+                )
+                assert response.status == 200
+                assert 1 == len(requests.results)
+                request_details = requests.results[0]
+                status, message = request_details.status, request_details.message
 
                 self._client.logger.info(
                     "Task %s creation status: %s (message=%s)",
                     self.id,
-                    status.state.value,
-                    status.message,
+                    status,
+                    message,
                 )
 
-                if (
-                    status.state.value
-                    == models.RqStatusStateEnum.allowed_values[("value",)]["FINISHED"]
-                ):
+                if "finished" == status.lower():
                     break
-                elif (
-                    status.state.value
-                    == models.RqStatusStateEnum.allowed_values[("value",)]["FAILED"]
-                ):
-                    raise exceptions.ApiException(
-                        status=status.state.value, reason=status.message, http_resp=response
-                    )
+                elif "failed" == status.lower():
```

Comment:
Probably, we need to check the statuses somehow. If their names are changed in the API, we can never know the SDK needs to be updated. I suggest that there is at least an assert or a test for this. The previous way of enum name access was far from convenient, but it also checked names automatically and would even be updated automatically after the schema changes.